### ISSUE TITLE
Error when instrumenting an "if" in an assembly block

### ISSUE THREAD

**Issue by chapati23** [sc-forks/solidity-coverage#287]
We're getting an error when trying to instrument one of our contracts that has some assembly:

`There was a problem instrumenting ./coverageEnv/contracts/MyContract.sol: SyntaxError: Expected "=:", "for", "function", "hex", "let", "return", "switch", "{", "}", comment, end of line, number, string, or whitespace but "i" found. Line: 29, Column: 7`

It fails in the `if` statement at the bottom.

Any idea why?

```solidity
  /// @notice Checks if a given address is whitelisted
  /// @return true if address is whitelisted, false if not
  function isWhitelisted
  (
    address _address
  )
    public
    view
    returns (bool _isWhitelisted)
  {
    bytes4 _signature = bytes4(keccak256("whitelisted(address)"));
    address _whitelistContract = getContractAddress("Whitelist");

    assembly {
      let _pointer := mload(0x40)  // Set _pointer to free memory pointer
      mstore(_pointer, _signature) // Store _signature at _pointer
      mstore(add(_pointer, 0x04), _address) // Store _address at _pointer. Offset by 4 bytes for previously stored _signature

      // staticcall(g, a, in, insize, out, outsize) => returns 0 on error, 1 on success
      let result := staticcall(
        gas,                // g = gas: whatever was passed already
        _whitelistContract, // a = address: _whitelist address assigned from getContractAddress()
        _pointer,           // in = mem in  mem[in..(in+insize): set to _pointer pointer
        0x24,               // insize = mem insize  mem[in..(in+insize): size of signature (bytes4) + bytes32 = 0x24
        _pointer,           // out = mem out  mem[out..(out+outsize): output assigned to this storage address
        0x20                // outsize = mem outsize  mem[out..(out+outsize): output should be 32byte slot (bool size = 0x01 < slot size 0x20)
      )

      // Revert if not successful
      if iszero(result) {
        revert(0, 0)
      }

      _isWhitelisted := mload(_pointer) // Assign result to returned value
      mstore(0x40, add(_pointer, 0x24)) // Advance free memory pointer by largest _pointer size
    }
  }
```

**Comment by cgewecke** [comment]
@chapati23 It looks like the parser needs to be updated to handle some of the newer assembly syntax.  Will fix this weekend, thanks for reporting.

**Comment by thekevinbrown** [comment]
Hey @cgewecke I'm hitting this issue too, wondering about how the fix is going. Any news? Anything I can do to help?

**Comment by 0age** [comment]
If you have the flexibility to change the inline assembly a bit, a workaround is to use a `switch` statement instead. So:
```
      if iszero(result) {
        revert(0, 0)
      }
```
becomes:
```
      switch iszero(result)
      case 1 {
        revert(0, 0)
      }
```

**Comment by patitonar** [comment]
is there any plan to address this issue?

**Comment by PaulRBerg** [comment]
Any updates on this? Happy to provide debug data.

**Comment by cgewecke** [comment]
This is finally fixed in `0.6.0`
